### PR TITLE
Remove deprecated decodedCacheRatioCap

### PR DESCRIPTION
**Pull request by mklim**
## Description

Remove deprecated decodedCacheRatioCap.

## Related Issues

flutter/flutter#26081

## Tests

I added the following tests:

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (Please read [Handling breaking changes]). https://groups.google.com/forum/#!topic/flutter-announce/xbYfKcpXbi0
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by liyuqian — APPROVED** [review]
LGTM. My only concern is that the announcement is sent today so we probably need to give developers more time to digest this breaking change before we merge it.

**Comment by mklim** [comment]
Sounds good. For what it's worth the field has also been marked as `@deprecated` in the framework for a few months now, so there's been much longer ongoing notice that way.

**Comment by cbracken** [comment]
@mklim thoughts on the timing of landing this? Given that it's been deprecated for a while now, is there any chance that we won't land this? If not, then I think it's reasonable to land now, as it'll take a bit to migrate up to beta channel in any case.

**Comment by mklim** [comment]
@cbracken I think that's reasonable. I'll land once the build is green.

**Pull request merged**
